### ISSUE TITLE
Default value is required to be present in the enum list

### ISSUE THREAD

**Issue by dejan** [linkedin/goavro#95]
Error `default value ought to encode using field schema` is thrown on trying to create codec (for decoding purposes) using schema with default values not present in the enum list. The Avro specification doesn't impose this restriction on the default value, yet this library can't cope with such schema.

Example schema:

```json
{
  "type": "record",
  "name": "person",
  "fields" : [
    {
      "name": "sex",
      "type": "enum",
      "default": "unknown",
      "symbols": [
        "male",
        "female"
      ]
    }
  ]
}
```

Error on `goavro.NewCodec` call:

> Record "person" field "sex": default value ought to encode using field schema: cannot encode binary enum "sex": value ought to be member of symbols: [male female]; "unknown"

**Comment by karrick** [comment]
The provided default value is "unknown", while the listed legal values for the enumeration only include "male" and "female". If "unknown" were listed as a legal value, then it would be fine.

The error message is trying to say that the provided default value must be encodable, and there is no way to encode "unknown" using the above schema.

`cannot encode binary enum "sex": value ought to be member of the symbols [male female]; "unkonwn"`.

In other words, "unknown" is not a member of the set ["male", "female"].

**Comment by dejan** [comment]
> The error message is trying to say that the provided default value must be encodable, and there is no way to encode "unknown" using the above schema.
> and there is no way to encode "unknown" using the above schema.

As I've explained the error happens **not on encoding** but on building the codec - for **decoding** purposes.

Let me provide you an use-case where this schema is useful. As you know, in order not to break backward compatibility, when adding a new field, one has to add a default value. Lets take the example field from my first message. Which default value would be a good choice if that was a field that's being added to an existing schema? Both "male" or "female" are wrong default values. It's also wrong to put "unknown" in the enum set as it would not force producer to put the actual value. 

Btw, official Java and C++ Avro libs work fine with this scenario. This one as well: https://github.com/elodina/go-avro

**Comment by karrick** [comment]
Based on my reading, the default value must be a legal value based on the schema's datum type.

Right now I'm leaning towards saying the schema is invalid simply because the provided default value is not one of the members of the list of legal values.

How would I recommend correcting it? By adding "unknown" to the list of legal enum values.
